cmd/pulumi.go

```diff
@@ -353,7 +353,7 @@ func getUpgradeCommand() string {
 		return "$ brew upgrade pulumi"
 	}
 
-	if filepath.Dir(exe) != filepath.Join(curUser.HomeDir, ".pulumi", "bin") {
+	if filepath.Dir(exe) != filepath.Join(curUser.HomeDir, workspace.BookkeepingDir, "bin") {
 		return ""
```

Comment:
Why bookkeepingdir here but homedir elsewhere?

Reply:
We would need to adjust the installation scripts first. This PR assumes the `bin` is still in its old place.